src/MongoDB.Driver.Core/Core/Connections/IsMasterHelper.cs

```diff
@@ -84,8 +99,8 @@ internal static CommandWireProtocol<BsonDocument> CreateProtocol(BsonDocument is
         {
             try
             {
-                var isMasterResult = await isMasterProtocol.ExecuteAsync(connection, cancellationToken).ConfigureAwait(false);
-                return new IsMasterResult(isMasterResult);
+                var isMasterResultDocument = await isMasterProtocol.ExecuteAsync(connection, cancellationToken).ConfigureAwait(false);
+                return new IsMasterResult(isMasterResultDocument);
```

Comment:
previously we didn't have below `catch` block for isMaster heartbeat calls (there were no catch blocks at all). Can it be a problem? If so, It can be simply fixed :)

Reply:
If I understand correctly, `ServerMonitor` previously did not use `IsMasterHelper.GetResultAsync` but now it is using it, and you're asking if the `catch` block will be a problem if called from the `ServerMonitor`? 

Since we shouldn't be sending _any_ authentication related information in the `ServerMonitor`'s `isMaster`, if we got an authentication error code 11, I'd imagine that would be a server bug. So I think what you have should be safe :)